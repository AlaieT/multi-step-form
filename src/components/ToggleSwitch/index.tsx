import React from "react";

import type { ToggleSwitchProps } from "../../types";

import styles from "../../styles/components/toggleSwitch.module.scss";

const ToggleSwitch = React.forwardRef<HTMLInputElement, ToggleSwitchProps>(
  ({ ...rest }, ref) => (
    <div className={styles.switch}>
      <input {...rest} ref={ref} type="checkbox" />
    </div>
  )
);

ToggleSwitch.displayName = "ToggleSwitch";

export default ToggleSwitch;
