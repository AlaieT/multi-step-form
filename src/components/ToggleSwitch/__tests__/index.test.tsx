import React from "react";
import { render, cleanup } from "@testing-library/react";

import ToggleSwitch from "..";

afterEach(cleanup);

describe("ToggleSwitch testing", () => {
  it("Snapshot testing", () => {
    const { asFragment } = render(<ToggleSwitch />);

    expect(asFragment()).toMatchSnapshot();
  });
});
