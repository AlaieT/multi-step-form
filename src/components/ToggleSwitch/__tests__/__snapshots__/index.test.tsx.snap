// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch testing Snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="switch"
  >
    <input
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;
